VideoLocker/src/main/java/org/edx/mobile/view/UserProfileFragment.java

```diff
@@ -0,0 +1,246 @@
+package org.edx.mobile.view;
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.text.TextUtils;
+import android.view.LayoutInflater;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.bumptech.glide.Glide;
+import com.bumptech.glide.RequestManager;
+import com.google.inject.Inject;
+
+import org.edx.mobile.R;
+import org.edx.mobile.model.api.ProfileModel;
+import org.edx.mobile.module.prefs.PrefManager;
+import org.edx.mobile.third_party.iconify.IconDrawable;
+import org.edx.mobile.third_party.iconify.Iconify;
+import org.edx.mobile.user.Account;
+import org.edx.mobile.user.GetAccountTask;
+import org.edx.mobile.util.ResourceUtil;
+
+import java.util.Locale;
+
+import roboguice.fragment.RoboFragment;
+import roboguice.inject.InjectExtra;
+
+public class UserProfileFragment extends RoboFragment {
+
+    @InjectExtra(UserProfileActivity.EXTRA_USERNAME)
+    private String username;
+
+    @Nullable
+    private GetAccountTask getAccountTask;
+
+    @Nullable
+    private Account account;
+
+    @Nullable
+    private ViewHolder viewHolder;
+
+    @Nullable
+    MenuItem editProfileMenuItem;
+
+    @Inject
+    private Router router;
+
+    private boolean isViewingOwnProfile;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setRetainInstance(true);
+        setHasOptionsMenu(true);
+
+        final ProfileModel model = new PrefManager(getActivity(), PrefManager.Pref.LOGIN).getCurrentUserProfile();
+        isViewingOwnProfile = null != model && model.username.equalsIgnoreCase(username);
+
+        getAccountTask = new GetAccountTask(getActivity(), username) {
+            @Override
+            protected void onSuccess(Account account) throws Exception {
```

Comment:
Sorry, looks like loading indication handling _is_ already handled via the `startProcess()` and `finishProcess()` methods of `TaskProcessCallback`, which is implemented by `BaseSingleFragmentActivity`. The design for setting it is not optimal though (it's mostly inferred from the `Context` parameter, and is kept as a weak reference).

There is also a `handleMessage()` implementation, but it has to be manually called from each `Task` subclass upon catching an exception. This could be changed to be triggered from the base `onException()` callback implementation instead. It is set up to trigger the `onMessage()` callback of it's `TaskProcessCallback` (which is also implemented in `BaseSingleFragmentActivity`.